src/components/calcite-button/calcite-button.tsx

```diff
@@ -123,6 +124,12 @@ export class CalciteButton {
       const elType = this.el.getAttribute("type");
       this.type = this.childElType === "button" && elType ? elType : "submit";
     }
+    this.initialOnClick = this.el.getAttribute("onclick");
```

Comment:
The only issue I see is if the `onClick` is added after initialization. 

Would this approach solve that issue? https://github.com/Esri/calcite-components/pull/317/files#diff-20c32e6a69324944338fbc027b1730d2R108